src/main/java/software/amazon/smithy/lsp/ext/SmithyCompletionItem.java

```diff
@@ -0,0 +1,63 @@
+package software.amazon.smithy.lsp.ext;
+
+import java.util.Optional;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.xtext.xbase.lib.Pure;
+
+import software.amazon.smithy.utils.Pair;
+
+public class SmithyCompletionItem {
+  private CompletionItem ci;
+  private Optional<Pair<String, String>> smithyImport = Optional.empty();
+
+  public SmithyCompletionItem(CompletionItem ci) {
+    this.ci = ci;
+  }
+
+  public SmithyCompletionItem(CompletionItem ci, String namespace, String id) {
+    this.ci = ci;
+    this.smithyImport = Optional.of(Pair.of(namespace, id));
+  }
+
+  public Optional<Pair<String, String>> getImport() {
+    return smithyImport;
+  }
+
+  public CompletionItem getCompletionItem() {
+    return ci;
+  }
+
+  public Optional<String> getQualifiedImport() {
+    return smithyImport.map(pair -> pair.left + "#" + pair.right);
+  }
+
+  public Optional<String> getImportNamespace() {
+    return smithyImport.map(f -> f.left);
+  }
+
+  @Override
+  @Pure
+  public boolean equals(final Object obj) {
```

Comment:
I miss case classes so, so much :'(

Reply:
Use Lombok? :)

Reply:
The thing that modifies bytecode for the sake of some getters?

**NO**

Reply:
In about 50 years I'll be able to upgrade this to Java 17: https://docs.oracle.com/en/java/javase/17/language/records.html